### PR TITLE
fix(api): consistent usage/documentation of hl_group

### PR DESCRIPTION
**Pull request by luukvbaal**
Problem:  Documentation for "hl_group" in nvim_buf_set_extmark() is
          unclear. "hl_group" in nvim_echo() does not accept
          highlight group id.
Solution: Move documentation for highlight group name/id to first
          mention of hl_group. Update nvim_echo() to accept highlight
          group id.

### CODE REVIEW

File: runtime/doc/api.txt [hunk 1]
```diff
@@ -2750,15 +2750,15 @@ nvim_buf_set_extmark({buffer}, {ns_id}, {line}, {col}, {opts})
...
-                  • sign_hl_group: name of the highlight group used to
-                    highlight the sign column text.
-                  • number_hl_group: name of the highlight group used to
-                    highlight the number column.
-                  • line_hl_group: name of the highlight group used to
-                    highlight the whole line.
-                  • cursorline_hl_group: name of the highlight group used to
-                    highlight the sign column text when the cursor is on the
-                    same line as the mark and 'cursorline' is enabled.
+                  • sign_hl_group: highlight group used to highlight the sign
```
Comment (bfredl) [inline runtime/doc/api.txt:2753]: repeating `highlight` sounds a bit unnattural, maybe just ` highlight group used for the sign column text`  (and so on for the others)

File: test/functional/api/vim_spec.lua [hunk 2]
```diff
@@ -3620,6 +3629,12 @@ describe('API', function()
         msg_a{15:msg_b}{16:msg_c}                         |
       ]],
       }
+      async_meths.nvim_echo({
+        { 'msg_a' },
+        { 'msg_b', api.nvim_get_hl_id_by_name('Statement') },
+        { 'msg_c', api.nvim_get_hl_id_by_name('Special') },
+      }, true, {})
+      screen:expect_unchanged()
```
Comment (bfredl) [inline test/functional/api/vim_spec.lua:3637]: nit: using `screen:expect_unchanged()` when a `screen:expect()` works fine (just needing some minor change) is considered an anti-pattern, as is it less robust than expecting a different state. 

In this case, better to change the messages slightly (like using `msg_d` or whatever), and then assert this distinct state.

Comment (bfredl) [reply test/functional/api/vim_spec.lua:3637]: I e `screen:unchanged()` is more for asserting that nothing happened at all UI-wise, which is not the case here. we are actively interested in a _different_ message.

Comment (luukvbaal) [reply test/functional/api/vim_spec.lua:3637]: Hmm yeah I knew it was intended to be used like that but I've also used it (here and in statuslcolumn/sign tests) as a way to test that different commands yield the same screen state. Adding new `screen:expect()` in such cases usually leads to warnings.

**Review by justinmk — APPROVED** [review]

**Comment by clason** [comment]
ASAN test was failing (and now fails on all PRs, preventing automerge 😞 )

**Comment by dundargoc** [comment]
I hate all of you. Stop merging PRs with failures lol.

**Comment by clason** [comment]
Good thing you already recorded your talk 😄

**Comment by luukvbaal** [comment]
Oopsie woopsie

**Pull request merged**